logstash-core/lib/logstash/runner.rb

```diff
@@ -149,8 +149,12 @@ def execute
       return 1
     end
 
-    # Print a warning to STDERR for bad java versions
-    LogStash::Util::JavaVersion.warn_on_bad_java_version
+    # Exit on bad java versions
+    java_version = LogStash::Util::JavaVersion.version
+    if LogStash::Util::JavaVersion.bad_java_version?(java_version)
+      $stderr.puts "Java version 1.8.0 or later is required. (You are running: #{java_version})"
+      return 1
+    end  
```

Comment:
We might want to delete the `warn_on_bad_java_version` since we wont use it anymore, IIRC we had also specs covering it.

https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/util/java_version.rb#L10